src/main/kotlin/com/vnea/intellij/plugin/movecursortoclass/action/MoveCursorToClassAction.kt

```diff
@@ -32,4 +33,8 @@ class MoveCursorToClassAction : AnAction() {
         }
 
     }
+
+    override fun getActionUpdateThread(): ActionUpdateThread {
+        return ActionUpdateThread.BGT
```

Comment:
**QUESTION**

Can you just explain me what is the purpose of this block of code?

Reply:
2022.3+ introduces this method with default return of BGT (preferred way) in `AnAction`. They added inspection saying that this has to be implemented and ask to do so in docs:
![image](https://user-images.githubusercontent.com/33603097/222979767-20e6a733-a99a-4ed7-a1d8-2e662758257f.png)
https://plugins.jetbrains.com/docs/intellij/working-with-custom-actions.html#creating-a-custom-action
![image](https://user-images.githubusercontent.com/33603097/222979788-4cd046d5-0873-4218-bbed-50f07feb5017.png)
https://plugins.jetbrains.com/docs/intellij/basic-action-system.html#principal-implementation-overrides
Since BGT provides all data in DataContext that is used here, I assumed it should be safe to return BGT (I have no experience with INtellij plugins :stuck_out_tongue: )
![image](https://user-images.githubusercontent.com/33603097/222979991-b958b9d6-a567-4a31-9754-6b7e210dfcfc.png)

Reply:
Okay I understand, thank you for the detailed answer. :)